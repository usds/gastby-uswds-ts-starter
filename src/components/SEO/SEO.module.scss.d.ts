declare namespace SEONamespace {
  export interface ISEOScss {
    sEOContainer: string;
  }
}

declare const SEOScssModule: SEONamespace.ISEOScss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: SEONamespace.ISEOScss;
};

export = SEOScssModule;
