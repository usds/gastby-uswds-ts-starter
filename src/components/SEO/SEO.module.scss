// @use "uswds-core" as *;

// .sEOContainer{

// }
