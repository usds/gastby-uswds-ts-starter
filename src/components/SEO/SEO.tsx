import React from 'react';
import { Helmet } from 'react-helmet';

// import * as styles from './SEO.module.scss';

export interface ISEOProps {
  children?: React.ReactNode;
  title: React.ReactNode;
}

const SEO = ({ title }: ISEOProps) => {
  // The reason why Gatsby HEAD API is not being used is because of:
  // https://github.com/gatsbyjs/gatsby/issues/36458#issuecomment-1334232043

  return (
    <Helmet defer={false}>
      <meta charSet="utf-8" />

      {/* Render Chrome if available or using latest version of Internet Explorer (Recommended). */}
      <meta httpEquiv="X-UA-Compatible" content="IE=edge; chrome=1" />

      <title>{`${title}`}</title>

      {/* Description content should ideally be between 160 - 200 characters */}
      <meta
        name="description"
        content={`This new web app will do X for the government providing Y, Z and W.`}
      />

      {/* Keyword content should list the top 10 words in order of relevance. The words chosen need
        to appear on every page. These words were chosen from the header as it appears on each page */}
      <meta
        name="keywords"
        content={`screening tool climate environmental economic justice beta council quality methodology`}
      />

      {/* Allows for Open Graph meta tags */}
      <meta property="og:url" content="https://new-usds-app.gov" />
      <meta property="og:title" content="New USDS App" />
      <meta property="og:site_name" content="New USDS App" />
      <meta
        property="og:description"
        content={`This new web app will do X for the government providing Y, Z and W.`}
      />

      {/* DAP Tag */}
      <script></script>
    </Helmet>
  );
};

export default SEO;
