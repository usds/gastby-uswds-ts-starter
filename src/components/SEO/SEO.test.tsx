import { render } from '@testing-library/react';
import { IntlProviderTestWrapper } from '@/test/testHelpers';
import SEO from './SEO';

describe(`rendering of SEO Component`, () => {
  const { asFragment } = render(
    <IntlProviderTestWrapper>
      <SEO title={`Page Title`} />
    </IntlProviderTestWrapper>,
  );
  it(`checks if component renders`, () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
