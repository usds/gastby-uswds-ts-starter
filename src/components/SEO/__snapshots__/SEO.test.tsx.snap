// Vitest Snapshot v1

exports[`rendering of SEO Component > checks if component renders 1`] = `<DocumentFragment />`;
